airflow/www/security.py

```diff
@@ -170,7 +190,11 @@ def __init__(self, appbuilder):
             if not view or not getattr(view, 'datamodel', None):
                 continue
             view.datamodel = CustomSQLAInterface(view.datamodel.obj)
+        app = self.appbuilder.get_app
         self.perms = None
+        # Custom cookie session interface
+        # Override to implement your custom cookie session interface
+        app.session_interface = DefaultSessionInterface()
```

Comment:
This shouldn't be set/change in here, but instead in a function in airflow.www.extensions.init_session (that you will need to call inside create_app in airflow.www.app

Reply:
Cool. But I'm of the opinion that users won't be able to modify the session when set at airflow.www.extensions.init_session?

Also, I think that some of the initialization at airflow.www.extensions can be refactored by moving them into the security manager. The security manager also gets called in create_app through airflow.www.extensions.init_appbuilder. 
So it's still the same I think but users can be able to override a lot using just the security manager

Reply:
The security manager is already in "god class" territory, so I think we should avoid moving anything else into it.

If we really want users to be able to customize everything (which I don't necessarily think is the case), we can easily allow customization of the init functions in other ways.

Reply:
Hi James, I discussed with Ash about this and the plans I have for also invalidating sessions.

I do not have strong opinion on this but having studied appbuider and that we also use it, I believe it's not antipattern and security manager is a good place to place security related components.

I would argue that what we did in airflow.www.extensions is actually an antipattern following how appbuider is built and also other flask packages.

Ash advised everything session related should be in one place and I would like if you can take more look at this. 

I'll be working on session invalidation and I will be moving it to security manager too if you do not object and have all session related components moved too to the security manager.

We can schedule a session and discuss this if you want.

Thanks!